src/registry/artifactory.rs

```diff
@@ -25,72 +29,99 @@ pub struct Artifactory {
     client: reqwest::Client,
 }
 
+#[derive(Error, Debug)]
+#[error("{0} is not a supported version requirement")]
+struct UnsupportedVersionRequirement(VersionReq);
+
 impl Artifactory {
-    pub fn new(registry: RegistryUri, credentials: &Credentials) -> eyre::Result<Self> {
+    /// Creates a new instance of an Artifactory registry client
+    pub fn new(registry: RegistryUri, credentials: &Credentials) -> miette::Result<Self> {
         Ok(Self {
             registry: registry.clone(),
             token: credentials.registry_tokens.get(&registry).cloned(),
             client: reqwest::Client::builder()
                 .redirect(reqwest::redirect::Policy::none())
-                .build()?,
+                .build()
+                .into_diagnostic()?,
         })
     }
 
-    /// Pings artifactory to ensure registry access is working
-    pub async fn ping(&self) -> eyre::Result<()> {
-        let repositories_url: Url = {
-            let mut uri = self.registry.to_owned();
-            let path = &format!("{}/api/repositories", uri.path());
-            uri.set_path(path);
-            uri.into()
-        };
-
-        let mut request = self.client.get(repositories_url);
+    async fn make_auth_request(
+        &self,
+        method: Method,
+        url: Url,
+        body: Option<Bytes>,
+    ) -> miette::Result<Response> {
+        let mut request_builder = self.client.request(method.clone(), url.clone());
```

Comment:
```suggestion
        let mut request = self.client.request(method, &url);
```

does this work?

Reply:
No, not with the reference, but the clones are not necessary (they're residual from PR #95) and I'll remove them.